test/indexer/test_hca_indexer.py

```diff
@@ -123,6 +124,44 @@ class BundleAndSize(NamedTuple):
                 finally:
                     self._delete_indices()
 
+    def test_duplicate_notification(self):
+        manifest, metadata = self._load_canned_bundle(self.new_bundle)
+        tallies: Counter = self._write_contributions(self.new_bundle, manifest, metadata)
```

Comment:
It was because then I got a type error for calling `update()` and `__setitem__()` (in test_zero_tallies). I failed to notice that it replaced one type error with another. I'll change to cast to dict.

Reply:
Type hints do nothing at runtime unless the code explicitly inspects them at runtime, which is rare. They affect tooling only:, IDEs, mypy etc. 

`dict` would also be wrong. You shouldn't need the hint at all because the called method explicitly declares the type as `Tallies` and PyCharm should have no problem inferring it. `Tallies` is an [alias](https://www.python.org/dev/peps/pep-0484/#type-aliases) I think, but that should be ok, too, if not it may be bug in PyCharm.

But again, type hints do not affect runtime behavior unless code explicitly inspects them. `typing.NamedTuple` or `dataclasses.dataclass` are examples of code that does. IOW, I am puzzled by your explanation.

Reply:
I was being imprecise in my last message when I said type error. I meant Pycharm's type complaints.

Tallies is a Mapping (not a MutableMapping) and therefore Pycharm complains for the calls to update() and __setitem__() [here](https://github.com/DataBiosphere/azul/pull/1036/files#diff-6dd72c70de92b16a622988cd255da8e3R133} and [here](https://github.com/DataBiosphere/azul/pull/1036/files#diff-6dd72c70de92b16a622988cd255da8e3R149).

Casting Tallies to dict give me an object with update() and setitem() defined which seems like a reasonable solution. Really, I only need to cast to MutableMapping in order to avoid the type complaints, but I couldn't figure out how to do that.

Reply:
I see. A declared return type of Mapping indicates the the caller should not modify the returned value. I think it is incidental though, before this change no call site modified the returned value so I chose the more restrictive Mapping, in violation of https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#15type-hints actually. The problem is that when `Tallies` is used for argument type hints, we **do** want to prohibit modification by the called method. So maybe the solution is to have a Tallies and a MutableTallies alias. Anyways, that's beyond the scope of this PR and I think copying into another dict is the right approach. Casting to `dict` or `Counter` was definitely the wrong approach.

```
tallies = {**tallies, **self._write_contributions(self.new_bundle, manifest, metadata)}
```

Reply:
… is what I would have done but the existing code is fine, too.